src/backend/mi2/mi2.ts

```diff
@@ -24,6 +24,50 @@ function couldBeOutput(line: string) {
 
 const trace = false;
 
+class LogMessage {
+	protected logMsgVar = "";
+	protected logMsgVarProcess = "";
+	protected logMsgRplNum = 0;
+	protected logMsgRplItem: string[] = [];
+	protected logMsgMatch = /(^\$[0-9]*[\ ]*=[\ ]*)(.*)/;
+	protected logReplaceTest = /{([^}]*)}/g;
+	public logMsgBrkList: Breakpoint[] = [];
+
+	logMsgOutput(record){
+		if ((record.type === 'console')) {
+			if(record.content.startsWith("$")){
+				const content = record.content;
+				const variableMatch = this.logMsgMatch.exec(content);
+				if (variableMatch) {
+					const value = content.substr(variableMatch[1].length).trim();
+					this.logMsgRplItem.push(value);
+					this.logMsgRplNum--;
+					if(this.logMsgRplNum == 0){
+						for(let i = 0; i < this.logMsgRplItem.length; i++){
+							this.logMsgVarProcess = this.logMsgVarProcess.replace("placeHolderForVariable", this.logMsgRplItem[i]);
+						}
+						return "Log Message:"  + this.logMsgVarProcess;
+					}
+				}
+			}
+			return null
+		}
+	}
+
+	logMsgProcess(parsed){
+		this.logMsgBrkList.forEach((brk)=>{
+			if(parsed.outOfBandRecord[0].output[0][1] == "breakpoint-hit" && parsed.outOfBandRecord[0].output[2][1] == brk.id){
+				this.logMsgVar = brk?.logMessage;
+				const matches = this.logMsgVar.match(this.logReplaceTest);
+				const count = matches ? matches.length : 0;
+				this.logMsgRplNum = count;
+				this.logMsgVarProcess = this.logMsgVar.replace(this.logReplaceTest, "placeHolderForVariable");
```

Comment:
I'm not a huge fan of the magic string here and would much prefer if we could instead maybe split by the regex, so it's not possible to have this value accidentally in the output message (even if very unlikely)